Modules/pyexpat.c

```diff
@@ -1839,37 +1848,40 @@ add_features(PyObject *mod)
 static int
 pyexpat_exec(PyObject *mod)
 {
-    if (PyType_Ready(&Xmlparsetype) < 0) {
-        return -1;
-    }
+    pyexpat_state *state = pyexpat_get_state(mod);
+    state->xml_parse_type = (PyTypeObject *)PyType_FromModuleAndSpec(
+        mod, &_xml_parse_type_spec, NULL);
 
-    if (init_handler_descrs() < 0) {
+    if (state->xml_parse_type == NULL) {
         return -1;
     }
 
-    /* Add some symbolic constants to the module */
-    if (ErrorObject == NULL) {
-        ErrorObject = PyErr_NewException("xml.parsers.expat.ExpatError",
-                                         NULL, NULL);
+    if (init_handler_descrs(state) < 0) {
+        return -1;
     }
-    if (ErrorObject == NULL) {
+    state->error = PyErr_NewException("xml.parsers.expat.ExpatError",
+                                      NULL, NULL);
+    if (state->error == NULL) {
         return -1;
     }
 
-    Py_INCREF(ErrorObject);
-    if (PyModule_AddObject(mod, "error", ErrorObject) < 0) {
-        Py_DECREF(ErrorObject);
+    /* Add some symbolic constants to the module */
+
+    Py_INCREF(state->error);
+    if (PyModule_AddObject(mod, "error", state->error) < 0) {
```

Comment:
Would you mind to use PyModule_AddObjectRef() here and below? You can remove the INCREF before and DECREF on error if you use PyModule_AddObjectRef().